### PR TITLE
Qualification tool - add in estimating the App end time when the event log missing application end event

### PR DESCRIPTION
**Pull request by tgravescs**
Many of the databricks event logs are missing the application end event so the application duration comes out as 0. To allow us to still score it, use the latest job end time or sql execution time instead and then add a column to indicate that we estimated the app duration in case user wants to know.

Note this does not change it for the profiling tool.

### CODE REVIEW

**Comment by tgravescs** [comment]
build

**Comment by tgravescs** [comment]
@gerashegalov new pr, I addressed your one comment. I plan to add more tests related to empty jobs/sql under https://github.com/NVIDIA/spark-rapids/issues/2652

**Review by gerashegalov — APPROVED** [review]

File: tools/src/main/scala/org/apache/spark/sql/rapids/tool/profiling/ApplicationInfo.scala [hunk 1]
```diff
@@ -320,15 +321,33 @@ class ApplicationInfo(
...
       val appStartNew: ArrayBuffer[ApplicationCase] = ArrayBuffer[ApplicationCase]()
       for (res <- this.appStart) {
-        val durationResult = ProfileUtils.OptionLongMinusLong(this.appEndTime, res.startTime)
+
+        val estimatedResult =
+          this.appEndTime match {
+            case Some(t) => this.appEndTime
+            case None =>
+              if (this.sqlEndTime.isEmpty && this.jobEndTime.isEmpty) {
+                None
```
Comment (nartal1) [inline tools/src/main/scala/org/apache/spark/sql/rapids/tool/profiling/ApplicationInfo.scala:330]: Just for my understanding. If both are empty then it throws NoSuchElementException?

**Review by nartal1 — APPROVED** [review]

File: tools/src/main/scala/org/apache/spark/sql/rapids/tool/profiling/ApplicationInfo.scala (hunk 1 above)
Comment (tgravescs) [reply tools/src/main/scala/org/apache/spark/sql/rapids/tool/profiling/ApplicationInfo.scala:330]: it returns None which is set to estimatedResult and that gets passed into OptionLongMinusLong which will just end up returning None for the durationResult which will result in null duration in table

Comment (nartal1) [reply tools/src/main/scala/org/apache/spark/sql/rapids/tool/profiling/ApplicationInfo.scala:330]: Thanks !
